Automattic/jetpack
# Donations: Send users back to editor after connecting to Stripe

https://github.com/Automattic/jetpack/issues/16888

#### Changes proposed in this Pull Request:

* This PR modifies the Stripe connection URL of the Donations block to make sure users are sent back to the editor once the Stripe connection is set up, so they can continue editing the page/post.
* It also removes the initial loading spinner, so users can start editing the block immediately. We still display an error placeholder if the API request fails. This is somehow unrelated to the issue this PR tries to solve, but decided to include here given it's a relatively small change.

#### Jetpack product discussion
p58i-9k2-p2
#### Does this pull request change what data or activity we track or use?
No.
#### Testing instructions:
* Set up a test site:
  * [Create a JN site running this branch](https://jurassic.ninja/create/?jetpack-beta&branch=update/donations-stripe-redirect-editor). 
  * Go to `/wp-admin/options-general.php?page=companion_settings`.
    * Activate `JETPACK_BETA_BLOCKS`. 
    * Set `JETPACK__SANDBOX_DOMAIN` with your sandbox address (make sure you enter the short URL e.g. mysandbox.wordpress.com to avoid issues with the SSL certificate).
  * Sandbox the API, the store and `subscribe.wordpress.com` (see PCYsg-lW4-p2 #sandbox-method for detailed instructions).
  * Set up Jetpack and purchase a Premium or Professional plan.
* Go to Posts > New and insert a new Donations block.
* Set up the Stripe connection (you will be able to see a link above the form letting you to skip and instantly create a test account).
* Make sure you're sent back to the editor after connecting to Stripe.
* Go to `wordpress.com/earn/payments/:site` and remove the Stripe connection.
* Go to Posts > New and insert a new Payments block.
* Set up the Stripe connection again.
* Make sure you're sent back again to the editor.

#### Proposed changelog entry for your changes:
N/A.
